Panic on startup (nil dereference) running on Nomad

I am running this as a CSI plugin to Nomad. I followed [this](https://github.com/seaweedfs/seaweedfs-csi-driver/tree/master/deploy/nomad) example, except 
- I use Nomad service discovery (not consul)
- Filer is using leveldb2 store not Postgres
- Master is single instance

The CSI plugin fails on any Nomad client (any pod) so I think the trace is not specific to the host machine, althoguh all my machines are configured very similarly. Version is `latest` for the CSI image, 3.55 for filer, volumes, master etc.

Logs:
```
I0807 23:46:18.075502 driver.go:105 starting
I0807 23:46:18.075881 server.go:94 Listening for connections on address: &net.UnixAddr{Name:"/csi/csi.sock", Net:"unix"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xcbb35e]

goroutine 72 [running]:
github.com/seaweedfs/seaweedfs-csi-driver/pkg/driver.(*ControllerServer).ControllerGetCapabilities(0x0, {0xc000125940?, 0x40da07?}, 0x10?)
	/go/src/github.com/seaweedfs/seaweedfs-csi-driver/pkg/driver/controllerserver.go:179 +0x5e
github.com/container-storage-interface/spec/lib/go/csi._Controller_ControllerGetCapabilities_Handler.func1({0x101cef0, 0xc0003f2ea0}, {0xe3f5e0?, 0xc000446200})
	/go/pkg/mod/github.com/container-storage-interface/spec@v1.8.0/lib/go/csi/csi.pb.go:6546 +0x78
github.com/seaweedfs/seaweedfs-csi-driver/pkg/driver.logGRPC({0x101cef0, 0xc0003f2ea0}, {0xe3f5e0, 0xc000446200}, 0xc000446220, 0xc0000a8318)
	/go/src/github.com/seaweedfs/seaweedfs-csi-driver/pkg/driver/utils.go:64 +0x132
github.com/container-storage-interface/spec/lib/go/csi._Controller_ControllerGetCapabilities_Handler({0xe6fe20?, 0x0}, {0x101cef0, 0xc0003f2ea0}, 0xc0002a4310, 0xf2ef48)
	/go/pkg/mod/github.com/container-storage-interface/spec@v1.8.0/lib/go/csi/csi.pb.go:6548 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000356000, {0x1021760, 0xc0003fa4e0}, 0xc000336360, 0xc00031b410, 0x168cba8, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000356000, {0x1021760, 0xc0003fa4e0}, 0xc000336360, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.57.0/server.go:980 +0x18c
```

CSI plugin job:
```hcl
job "seaweedfs-plugin" {
  datacenters = ["dc1"]
  type        = "system"
  update {
    max_parallel = 1
    stagger      = "60s"
  }

  # only one plugin of a given type and ID should be deployed on
  # any given client node
  constraint {
    operator = "distinct_hosts"
    value    = true
  }

  group "nodes" {
    ephemeral_disk {
      migrate = false
      size    = 5000
      sticky  = false
    }
    restart {
      interval = "5m"
      attempts = 10
      delay    = "15s"
      mode     = "delay"
    }
    # does not need to run on a client with seaweed, only needs docker privileged
    task "plugin" {
      driver = "docker"

      template {
        destination = "config/.env"
        change_mode = "restart"
        env         = true
        data        = <<-EOF
{{ range $i, $s := nomadService "seaweedfs-filer-http" }}
{{- if eq $i 0 -}}
SEAWEEDFS_FILER_IP_http={{ .Address }}
SEAWEEDFS_FILER_PORT_http={{ .Port }}
{{- end -}}
{{ end }}
{{ range $i, $s := nomadService "seaweedfs-filer-grpc" }}
{{- if eq $i 0 -}}
SEAWEEDFS_FILER_IP_grpc={{ .Address }}
SEAWEEDFS_FILER_PORT_grpc={{ .Port }}
{{- end -}}
{{ end }}
EOF
      }

      config {
        network_mode = "host"
        image        = "chrislusf/seaweedfs-csi-driver:latest"
        force_pull   = "true"

        args = [
          "--endpoint=unix://csi/csi.sock",
          "--filer=${SEAWEEDFS_FILER_IP_http}:${SEAWEEDFS_FILER_PORT_http}.${SEAWEEDFS_FILER_PORT_grpc}",
          "--nodeid=${node.unique.name}",
          "--cacheCapacityMB=1000",
          "--cacheDir=${NOMAD_TASK_DIR}/cache_dir",
        ]

        privileged = true
      }

      csi_plugin {
        id        = "seaweedfs"
        type      = "monolith"
        mount_dir = "/csi"
      }
      resources {
        cpu        = 100
        memory     = 512
        memory_max = 2048
      }
    }
  }
}
```

Let me know if I should provide more info.

--- a/cmd/seaweedfs-csi-driver/main.go
+++ b/cmd/seaweedfs-csi-driver/main.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"strings"
 
 	"github.com/seaweedfs/seaweedfs-csi-driver/pkg/datalocality"
 	"github.com/seaweedfs/seaweedfs-csi-driver/pkg/driver"
@@ -12,9 +13,8 @@ import (
 )
 
 var (
-	runNode        = flag.Bool("node", false, "run node server")
-	runController  = flag.Bool("controller", false, "run controller server")
-	enableAttacher = flag.Bool("attacher", false, "enable attacher")
+	components     = flag.String("components", "controller,node", "components to run, by default both controller and node")
+	enableAttacher = flag.Bool("attacher", true, "enable attacher, by default enabled for backward compatibility")
 
 	filer             = flag.String("filer", "localhost:8888", "filer server")
 	endpoint          = flag.String("endpoint", "unix://tmp/seaweedfs-csi.sock", "CSI endpoint to accept gRPC calls")
@@ -55,12 +55,32 @@ func main() {
 		os.Exit(1)
 	}
 
+	runNode := false
+	runController := false
+	for _, c := range strings.Split(*components, ",") {
+		switch c {
+		case "controller":
+			runController = true
+		case "node":
+			runNode = true
+		default:
+			glog.Errorf("invalid component: %s", c)
+			os.Exit(1)
+		}
+	}
+
+	glog.Infof("will run node: %v, controller: %v, attacher: %v", runNode, runController, *enableAttacher)
+	if !runNode && !runController {
+		glog.Errorf("at least one component should be enabled: either controller or node (use --components=...)")
+		os.Exit(1)
+	}
+
 	glog.Infof("connect to filer %s", *filer)
 
 	drv := driver.NewSeaweedFsDriver(*filer, *nodeID, *endpoint, *enableAttacher)
 
-	drv.RunNode = *runNode
-	drv.RunController = *runController
+	drv.RunNode = runNode
+	drv.RunController = runController
 
 	drv.ConcurrentWriters = *concurrentWriters
 	drv.CacheCapacityMB = *cacheCapacityMB

--- a/deploy/helm/seaweedfs-csi-driver/Chart.yaml
+++ b/deploy/helm/seaweedfs-csi-driver/Chart.yaml
@@ -2,5 +2,5 @@ apiVersion: v2
 name: seaweedfs-csi-driver
 description: A Helm chart for Kubernetes CSI backed by a SeaweedFS cluster
 type: application
-version: 0.2.0
+version: 0.2.1
 appVersion: latest

--- a/deploy/helm/seaweedfs-csi-driver/templates/daemonset.yaml
+++ b/deploy/helm/seaweedfs-csi-driver/templates/daemonset.yaml
@@ -45,7 +45,7 @@ spec:
             {{- if .Values.node.injectTopologyInfoFromNodeLabel.enabled }}
             - --dataCenter=$(DATACENTER)
             {{- end }}
-            - --node
+            - --components=node
           env:
             - name: CSI_ENDPOINT
               value: unix:///csi/csi.sock

--- a/deploy/helm/seaweedfs-csi-driver/templates/deployment.yaml
+++ b/deploy/helm/seaweedfs-csi-driver/templates/deployment.yaml
@@ -44,10 +44,8 @@ spec:
             - --endpoint=$(CSI_ENDPOINT)
             - --filer=$(SEAWEEDFS_FILER)
             - --nodeid=$(NODE_ID)"
-            - --controller
-            {{- if .Values.csiAttacher.enabled }}
-            - --attacher
-            {{- end }}
+            - --components=controller
+            - --attacher={{ .Values.csiAttacher.enabled }}
           env:
             - name: CSI_ENDPOINT
               value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock

